### PR TITLE
Addoncontroller: Report errors on cluster object

### PR DESCRIPTION
**Pull request by alvaroaleman**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
none
```

/assign @mrIncompetent

### CODE REVIEW

File: api/pkg/controller/addon/addon_controller.go [hunk 1]
```diff
@@ -143,6 +143,19 @@ func (r *Reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
...
+		if err := r.List(ctx, &ctrlruntimeclient.ListOptions{}, clusterList); err != nil {
+			glog.Errorf("failed to list clusters to find cluster to report error to: %v", err)
+		}
+		for _, cluster := range clusterList.Items {
+			if cluster.Status.NamespaceName == addon.Namespace {
+				r.recorder.Eventf(&cluster, corev1.EventTypeWarning, "ReconcilingError",
+					"failed to reconcile Addon %q: %v", addon.Name, reconcilingError)
+				break
+			}
+		}
```
Comment (mrIncompetent) [inline api/pkg/controller/addon/addon_controller.go:154]: Use `addon.Spec.Cluster` instead

Comment (alvaroaleman) [reply api/pkg/controller/addon/addon_controller.go:154]: Updated, PTAL

**Pull request merged**